research/src/pages/component-spec-template.mdx

```diff
@@ -0,0 +1,153 @@
+> This document represents a component specification template. Its purpose
+> is to provide a point-in-time guide as to how a component should be
+> authored, in light of common anatomies, behaviors, and accessibility.
+> The primary audience is a developer authoring a given component.
+
+# {ComponentName} component specification
+
+> Enter a description here. Example:
+>
+> The `{ComponentName}` component allows a user to slide a single thumb along a horizontal or vertical axis, representing a min/max range.
+
+## Visual examples
+
+> Provide, if possible, visual examples of the component being built, to provide an idea of implementations.
+
+## DOM hierarchy/structure
+
+> Define the recommended DOM shape to represent the component, and how important attributes (aria attributes especially) are applied to the various parts. In cases where a component nests other components, ideally, you expand the full dom structure to understand both the expectation and if there are shortcommings in the child components' customizability.
+>
+> Example:
+>
+> ```html
+> <div class="root">
+>   <div class="rail"></div>
+>
+>   <div class="mark">
+>     <label class="markLabel" />
+>   </div>
+>   <div class="track" />
+>   <div
+>     class="thumb"
+>     tabindex="0"
+>     role="slider"
+>     aria-valuenow="0"
+>     aria-valuemin="0"
+>     aria-valuemax="10"
+>   />
+>   <input name="{name}" type="hidden" value="0" />
+> </div>
+> ```
+
+## Component anatomy
+
+> Outline the parts of a component; what their names and purposes are. Ideally order them in DOM order. Also consider if the component should be broken into subcomponents. Example:
+>
+> | Name        | Considerations                                                       |
+> | ----------- | -------------------------------------------------------------------- |
+> | `root`      | The root element.                                                    |
+> | `rail`      | The line behind the Slider thumb and rail.                           |
+> | `mark`      | Optional mark line on top of the rail, below the track.              |
+> | `markLabel` | A label aligned with a mark to render the textual value of the mark. |
+> | `track`     | The selected area of the Slider.                                     |
+> | `thumb`     | The focusable draggable knob within the Slider.                      |
+> | `tooltip`   | The tooltip rendered above a thumb.                                  |
+
```

Comment:
Would add a subsection here for events at the anatomy level (which are distinct from the input-specific events like onClick or onKeyDown, already included below).

For example, in the WIP `<select>` spec, one event is "`::host` Listening for `toggle-pop-up` event to hide/show `::part-pop-up`."